### PR TITLE
Integrated scarf sdk

### LINKED ISSUE #612: Integrate Scarf SDK on NPM
Integrate scarf sdk with `@tailcallhq/tailcall`.

**Additional Info**
https://www.npmjs.com/package/@scarf/scarf

### PR DESCRIPTION
**Pull request by Min2who**
fixed #612 
/claim #612

### CODE REVIEW

File: npm/gen-root.ts [hunk 1]
```diff
@@ -47,8 +47,13 @@ async function genServerPackage(buildDefinitions: string[]) {
         type: 'module',
         version: packageVersion,
         optionalDependencies,
+        scarfSettings: {
+            defaultOptIn: true
```
Comment (tusharmath) [inline npm/gen-root.ts:51]:
```suggestion
            defaultOptIn: true,
            allowTopLevel: true
```

Comment (Min2who) [reply npm/gen-root.ts]: @tusharmath i have made the changes please review it

**Comment by amitksingh1490** [comment]
Please ensure thorough testing of the code, as the current implementation fails to generate a `package.json` and results in an error.

**Comment by Min2who** [comment]
> Please ensure thorough testing of the code, as the current implementation fails to generate a `package.json` and results in an error.

I have fixed it

**Comment by Min2who** [comment]
@tusharmath i have run the formatter and lint check

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**